fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -296,6 +303,55 @@ public DataApiRequest(
     }
 
     /**
+     * Update sort field string by truncating the dateTime if it is present.
+     *
+     * @param sorts  String of sort columns
+     *
+     * @return Updated sort column if dateTime exists. Otherwise return as it is received.
+     */
+    protected String truncateTimeSort(String sorts) {
+
+        if (sorts != null && !sorts.isEmpty() && sorts.contains(DATE_TIME_STRING)) {
+            if (sorts.contains(",")) {
+                //Truncate the dateTime sort column from the api sort string
+                return sorts.substring(sorts.indexOf(",") + 1, sorts.length());
+            } else {
+                //If only dateTime column requested for sorting, then rest of the api sort value will be null
+                return null;
+            }
+        } else {
+            //If there is no dateTime string involved, return as it is received
+            return sorts;
+        }
+    }
+
+    /**
+     * Generate OrderByColumn instance for dateTime sort column if it is present in string of sort columns.
+     *
+     * @param sorts  String of sort columns
+     *
+     * @return instance of OrderByColumn for dateTime
+     */
+    protected Optional<OrderByColumn> generateDateTimeSortColumn(String sorts) {
+
+        if (sorts != null && !sorts.isEmpty() && sorts.contains(DATE_TIME_STRING)) {
+            //Requested sort on dateTime - It has to be the first field in string
+            String dateTime = sorts.contains(",") ? sorts.substring(0, sorts.indexOf(",")) : sorts;
+            List<String> dateTimeWithDirection = Arrays.asList(dateTime.split("\\|"));
```

Comment:
This here seems like work that we're duplicating with the "normal" sort processing. Again, can we not duplicate the work?

One idea would be to generate the sort _objects_ (as a list) for the entire clause (so this method will need to handle dateTime not being a column but just a String), and then process that list of sort objects to pull off the dateTime sort from the front, and otherwise validate the sorts.